### PR TITLE
Add high-level API actions for POA&Ms, SAPs, and SARs

### LINKED ISSUE #56: Add high-level API actions for POA&Ms, SAPs, and SARs
Today, the API specification has support for Catalogs, Profiles, Component Definitions, and SSPs. Some of these have support for "deeper" routes (for example, `/component-definitions/{componentDefinitionId}/parties/{partyId}`) but other document types, we keep it a little more high-level (for now), such as the routes under `/catalogs/`. Adding support for additional document types at a similar level of granularity as a Catalog will at least provide basic support for creating, reading, updating, deleting, and listing these other documents.

### PR DESCRIPTION
**Pull request by tuckerzp**
This will close #56

### CODE REVIEW

**Comment by tuckerzp** [comment]
Since I will almost certainly miss some things we should add the the api specs, I have made this a draft so people can leave there opinions as I add things.

File: openapi.yaml [hunk 1]
```diff
@@ -2165,6 +2177,634 @@ paths:
       security:
         - oscal_auth:
             - read:parties
+  /plan-of-action-and-milestones:
+    get:
+      tags:
+        - OSCAL Plan of Action and Milestones
+      summary: Returns all OSCAL Plan of Action and Milestones
```
Comment (laurelmay) [inline openapi.yaml:2184]: @brian-easyd Can you grammar check here? "OSCAL Plan of Action and Milestones" or "OSCAL Plans of Action and Milestones" (like "attorneys general")?

File: openapi.yaml [hunk 2]
```diff
@@ -2165,6 +2177,634 @@ paths:
...
+                items:
+                  $ref: "#/components/schemas/OSCALCatalog"
+        400:
+          description: Invalid status value
+          content: {}
+      security:
+        - oscal_auth:
+            - read:catalogs
+
+  /system-assessment-plans:
```
Comment (nuttercd) [inline openapi.yaml:2389]: Swagger Editor is showing a structural error here, and I'm not sure why. This is also happening on line 2598.

File: openapi.yaml [hunk 3]
```diff
@@ -2165,6 +2177,634 @@ paths:
...
+      x-codegen-request-body-name: body
+    /system-assesment-plans/{catalogId}:
+      get:
+      tags:
+        - OSCAL System Assessment Plans
+      summary: Finds an OSCAL System Assessment Plans catalog by ID
+      description: Returns a single OSCAL System Assessment Plan
+      operationId: getAssessmentPlanById
+      parameters:
+        - name: catalogId
```
Comment (nuttercd) [inline openapi.yaml]: Swagger Editor is also giving errors on corresponding `{catelogId}` in the `/system-assessment-plans` path. Is this not already part of the path as shown in line 2439 `/system-assessment-plans/{catalogId}`?

Comment (tuckerzp) [reply openapi.yaml]: Can you post the actual error?

File: openapi.yaml (hunk 1 above)
Comment (tuckerzp) [reply openapi.yaml:2184]: Based on https://pages.nist.gov/OSCAL/reference/latest/complete/json-outline/, NIST has it as OSCAL Plan of Action and Milestones. But agree `Plans` may read better, especially on line `2184`.

File: openapi.yaml (hunk 3 above)
Comment (nuttercd) [reply openapi.yaml]: "Semantic error at paths./system-assessment-plans./system-assesment-plans/{catalogId}.parameters.0.name
Path parameter "catalogId" must have the corresponding {catalogId} segment in the "/system-assessment-plans" path"

Comment (tuckerzp) [reply openapi.yaml]: Would an `getAssessmentPlanById` even use a `catalogId`? They may be a remnant of me coping the data from `Catalog` and need to be updated with correct info.

File: openapi.yaml (hunk 2 above)
Comment (nuttercd) [reply openapi.yaml:2389]: "Structural error at paths./system-assessment-plans should NOT have additional properties additionalProperty: /system-assesment-plans/{catalogId}, /system-assesment-plans/search"

File: openapi.yaml (hunk 1 above)
Comment (laurelmay) [reply openapi.yaml:2184]: Yeah but I think that situation, it's talking about a single POA&M (hence, `poam-items` as an attribute). If we're returning multiple POA&Ms then are there Plans?

**Comment by tuckerzp** [comment]
Closing this pr so that I can actually review the changes.

File: openapi.yaml (hunk 1 above)
Comment (brian-ruf-ezd) [reply openapi.yaml:2184]: @kylelaker this is correct as-is. When expanding the acronym, it officially has "Plan" as singular and "Milestones" as plural. The POA&M -is- the one and only plan for resolving vulnerabilities within a given system. It contains multiple milestones toward achieving that plan.

**Pull request closed without merging**